src/app/pages/swap/hooks/use-sbtc-deposit-transaction.tsx

```diff
@@ -129,34 +132,28 @@ export function useSbtcDepositTransaction(btcSigner?: Signer<P2Ret>) {
         outputs.forEach(output => {
           // Add change output
           if (!output.address) {
-            deposit.transaction.addOutputAddress(
-              btcSigner.address,
-              BigInt(output.value),
-              networkMode
-            );
+            deposit.transaction.addOutputAddress(signer.address, BigInt(output.value), networkMode);
             return;
           }
         });
 
-        return { deposit, fee, maxSignerFee };
+        return { deposit, fee: createMoney(fee, 'BTC') };
       } catch (error) {
         logger.error('Error generating deposit transaction', error);
         return null;
       }
     },
-    async onDepositSbtc(swapSubmissionData: SwapSubmissionData) {
-      if (!stacksAccount || !btcSigner) return;
-      const sBtcDeposit = swapSubmissionData.txData?.deposit as SbtcDeposit;
-
+    async onDepositSbtc(deposit?: SbtcDeposit) {
+      if (!deposit) return;
       try {
-        btcSigner.sign(sBtcDeposit.transaction);
-        sBtcDeposit.transaction.finalize();
-        logger.info('Deposit', { deposit: sBtcDeposit });
+        signer.sign(deposit.transaction);
```

Comment:
Can this be signed by software wallets only? Why doesn't it use our tx signer `useSignBitcoinTx`?

Reply:
Yeah, I had a note to refactor with Ledger integration bc I just didn't realizd that should be used. I'll look at changing here.

Reply:
I ended leaving for now bc Ledger still hits this issue. I'll have to address in the follow up issue to get Ledger working with sBTC deposits...

![Screenshot 2025-01-08 at 1 22 39 PM](https://github.com/user-attachments/assets/935e3537-7e2b-45a0-9bfc-76ce32e42624)